src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/SecureStringHelpers.cs

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Security;
+
+namespace Internal.Cryptography
+{
+    internal static class SecureStringHelpers
+    {
+        /// <summary>
+        /// Convert a SecureString or string into an IntPtr zero terminated Unicode string.
+        /// The returned IntPtr must be freed with Marshal.ZeroFreeGlobalAllocUnicode.
+        /// </summary>
+        internal static IntPtr PasswordToGlobalAllocUnicode(object password)
+        {
+            return PasswordToGlobalAlloc(password, true);
+        }
+
+        /// <summary>
+        /// Convert a SecureString or string into an IntPtr zero terminated ANSI string.
+        /// The returned IntPtr must be freed with Marshal.ZeroFreeGlobalAllocAnsi.
+        /// </summary>
+        internal static IntPtr PasswordToGlobalAllocAnsi(object password)
+        {
+            return PasswordToGlobalAlloc(password, false);
+        }
+
+        private static IntPtr PasswordToGlobalAlloc(object password, bool unicode)
+        {
+            if (password != null)
+            {
+                string passwordString = password as string;
+                if (passwordString != null)
+                {
+                    if (unicode)
+                    {
+                        return Marshal.StringToHGlobalUni(passwordString);
+                    }
+                    else
+                    {
+                        return Marshal.StringToHGlobalAnsi(passwordString);
+                    }
+                }
+
+                SecureString secureString = password as SecureString;
+                if (secureString != null)
+                {
+                    if (unicode)
+                    {
+                        return SecureStringMarshal.SecureStringToGlobalAllocUnicode(secureString);
+                    }
+                    else
+                    {
+                        return SecureStringMarshal.SecureStringToGlobalAllocAnsi(secureString);
+                    }
+                }
+            }
```

Comment:
`Debug.Fail($"Don't know how to marshal a password of type {secureString.GetType().FullName}");`?